src/Database/LSMTree/Internal/Run.hs

```diff
@@ -177,13 +195,21 @@ fromMutable :: HasFS IO h
             -> RefCount
             -> RunBuilder (FS.Handle h)
             -> IO (Run (FS.Handle h))
-fromMutable fs hbio caching refCount builder = do
+fromMutable fs hbio runRunDataCaching refCount builder = do
     (runRunFsPaths, runFilter, runIndex, runNumEntries) <-
       Builder.unsafeFinalise fs builder
     runRefCount <- newIORef refCount
     runKOpsFile <- FS.hOpen fs (runKOpsPath runRunFsPaths) FS.ReadMode
     runBlobFile <- FS.hOpen fs (runBlobPath runRunFsPaths) FS.ReadMode
-    setRunDataCaching hbio runKOpsFile caching
+    -- always drop filter, index and checksum files from the cache
+    FS.hDropCacheAll' fs hbio (runFilterPath runRunFsPaths)
+    FS.hDropCacheAll' fs hbio (runIndexPath runRunFsPaths)
+    FS.hDropCacheAll' fs hbio (runChecksumsPath runRunFsPaths)
```

Comment:
It'd be a bit neater to do this by doing it inside `Builder.unsafeFinalise` just before closing all the files. That is more local, and it would avoid having to re-open the files just to drop caches.

Also, it'd avoid any use of the `'` "primed versions of the hadvice utilities (the ones that open the file just to apply advice), which isn't a great pattern to encourage.